controllers/nginx/rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -236,7 +236,19 @@ http {
         ssl_verify_client on;
         ssl_verify_depth {{ $location.CertificateAuth.ValidationDepth }};
         {{ end }}
+        
+        {{ if (or $location.Redirect.ForceSSLRedirect (and (not (empty $server.SSLCertificate)) $location.Redirect.SSLRedirect)) }}
```

Comment:
@rikatz I was thinking something like this (substituting in appropriate ssl-wildcard secret):
```
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: default-route
  annotations:
    kubernetes.io/ingress.class: "nginx"
    kubernetes.io/ssl-redirect: "true"
spec:
  tls:
    - hosts:
      - "*.example.com"
      secretName: "ssl-wildcard"
  rules:
  - host: "www.example.com"
    http:
      paths:
      - path: /
        backend:
          serviceName: route-default
          servicePort: 80
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: check-route
  annotations:
    kubernetes.io/ingress.class: "nginx"
    ingress.kubernetes.io/ssl-redirect: "false"
spec:
  rules:
  - host: "www.example.com"
    http:
      paths:
      - path: /check
        backend:
          serviceName: route-check
          servicePort: 80
```

My thought is this should create something like:
```
 server {
    server_name www.example.com;
    ...
    location / {
        if ($pass_access_scheme = http) {
            return 301 https://$host$request_uri;
        }
        ...
    }
    location /check {
        ...
    }
```

Instead this code would generate (which always does ssl redirect):
```
 server {
    server_name www.example.com;
    ...
    if ($pass_access_scheme = http) {
        return 301 https://$host$request_uri;
    }
    location / {
        ...
    }
    location /check {
        ...
    }
```